### PR TITLE
Airport challenge 2 - Emanuele Gorga

### PR DESCRIPTION
**Pull request by emanuelegorga**
With a little better understanding I managed to review all my code and redo the challenge.

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @airport = []
+    @capacity = capacity
+  end
+
+  def land(plane)
+    raise "No planes can land, stormy weather." if @weather.stormy?
+    raise "No more planes can land, airport full." if full?
+    @airport << plane
+  end
+
```
Comment (CelineKaslin) [inline lib/airport.rb]: great use of the raise 👍

File: spec/airport_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  it { is_expected.to respond_to(:land).with(1).argument }
+
+  it { is_expected.to respond_to(:takeoff) }
+
+  it { is_expected.to respond_to(:full?) }
+
+  describe '#take off' do
+    it "prevents to take off in case of stormy" do
+      forecast = create_sunny_weather
```
Comment (CelineKaslin) [inline spec/airport_spec.rb]: I do not understand your forecast variable. What are you trying to do with it

File: spec/airport_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  it "has capacity of 20" do
+    expect(Airport::DEFAULT_CAPACITY).to eq 20
+  end
+
+  def create_weather(stormy)
+    weather = double("Weather")
+    allow(weather).to receive(:stormy?).and_return(stormy)
+    weather
+  end
```
Comment (CelineKaslin) [inline spec/airport_spec.rb]: I am not sure to understand why you are passing stormy as an argument of `and_return` and then weather. Do you want to return "stormy" as a string ? 
i might just do not know a syntax and just not understand it

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  def create_stormy_weather
+    create_weather(true)
+  end
+
+  def create_sunny_weather
+    create_weather(false)
+  end
+
```
Comment (CelineKaslin) [inline spec/airport_spec.rb]: for the 2 previous test, do you try to test or to create your method with a def ? 
Not sure that you are suppose to create a method in a spec.

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  it "has capacity of 20" do
+    expect(Airport::DEFAULT_CAPACITY).to eq 20
+  end
+
+  def create_weather(stormy)
+    weather = double("Weather")
+    allow(weather).to receive(:stormy?).and_return(stormy)
+    weather
+  end
+
```
Comment (scass91) [inline spec/airport_spec.rb]: I might be misreading this - is the spec file here creating 2 weather methods for stormy & sunny weather? Could these be stored within weather.rb?

File: spec/weather_spec.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+require 'weather'
+
+RSpec.describe Weather do
+
+  it { is_expected.to respond_to(:stormy?) }
+
+  it 'returns 1 or 2' do
+    expect(subject.stormy?).to be(1).or be(2)
+  end
```
Comment (CelineKaslin) [inline spec/weather_spec.rb]: great to test that it return what you want :) 
Maybe consider to stub to test if it return whatever you assign to one and whatever you assign to 2

File: spec/airport_spec.rb [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      allow(forecast).to receive(:stormy?).and_return(true)
+      expect { subject.land(airplane) }.to raise_error "No planes can land, stormy weather."
+    end
+  end
+
+  describe '#full?' do
+    it "prevents to land planes in case of full airport" do
+      capacity = 2
+      gatwick = Airport.new(create_sunny_weather, capacity)
+      gatwick.land(airplane)
```
Comment (scass91) [inline spec/airport_spec.rb]: Repeating here - consider changing this to 2 times.do - DRY

Comment (scass91) [reply spec/airport_spec.rb]: As in 2.times do Gatwick.land(airplane)

**Pull request closed without merging**